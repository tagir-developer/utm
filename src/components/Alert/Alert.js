import React from 'react'
import { CSSTransition } from 'react-transition-group'

const Alert = ({ alert, alertClose, copied }) => {

	return (
		<CSSTransition
			in={alert.visible && !copied}
			timeout={{
				enter: 600,
				exit: 200
			}}
			classNames='alert'
			mountOnEnter
			unmountOnExit
		>
			<div className={`alert alert-${alert.payload.type} alert-dismissible fade show`} role="alert">
				<strong>{alert.payload.heading}</strong>&nbsp;
			{alert.payload.message}
				<button
					type="button"
					className="close"
					onClick={alertClose}
				>
					<span aria-hidden="true">&times;</span>
				</button>
			</div>
		</CSSTransition>

	)
}

Alert.defaultProps = {
	alert: {
		visible: false,
		payload: {
			type: 'success',
			heading: 'Заголовок алерта',
			message: 'Сообщение алерта',
		}
	},
	alertClose: () => {},
	copied: false
}

export default Alert