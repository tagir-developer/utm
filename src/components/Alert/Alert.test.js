import React from 'react'
import Alert from './Alert'


describe('<Alert />', () => {

	const testAlertState = (aVisible, aType = 'success', aHeading = 'Алерт отображается', aMessage = 'Сообщение алерта' ) => ({
		visible: aVisible,
		payload: {
			type: aType,
			heading: aHeading,
			message: aMessage,
		}
	})

	let wrapper

	beforeEach(() => {
		wrapper = shallow(<Alert />)
	})

	describe('Component render. Component props testing', () => {

		it('Should Alert component render without props', () => {
			expect(toJson(wrapper)).toMatchSnapshot()
		})

		it('Should alert show', () => {
			wrapper.setProps({
				alert: testAlertState(true)
			})
			expect(wrapper.find('CSSTransition').props().in).toEqual(true)
		})

		it('Should alert hide', () => {
			wrapper.setProps({
				alert: testAlertState(false)
			})
			expect(wrapper.find('CSSTransition').props().in).toEqual(false)
		})

		it('Should render alert type', () => {
			wrapper.setProps({
				alert: testAlertState(true, 'danger')
			})
			expect(wrapper.find('div[role="alert"]').props().className).toEqual('alert alert-danger alert-dismissible fade show')
		})

		it('Should render alert heading', () => {
			wrapper.setProps({
				alert: testAlertState(true, 'success', 'Тестовый заголовок алерта')
			})
			expect(wrapper.find('strong').text()).toBe('Тестовый заголовок алерта')
		})

		it('Should render alert message', () => {
			wrapper.setProps({
				alert: testAlertState(true, 'success', 'Заголовок', 'Тестовое сообщение алерта')
			})
			expect(wrapper.find('div[role="alert"]').text()).toContain('Тестовое сообщение алерта')
		})

		it('Should alert hide if copied=true', () => {
			wrapper.setProps({
				copied: true
			})
			expect(wrapper.find('CSSTransition').props().in).toEqual(false)
		})

	})

	describe('Alert component handlers testing', () => {

		it('Should call "alertClose" handler', () => {
			const clickHandler = jest.fn()
			wrapper.setProps({
				alertClose: clickHandler
			})
			wrapper.find('button.close').simulate('click')
			expect(clickHandler).toHaveBeenCalled()
		})

		it('Should use default "alertClose" handler', () => {
			const result = Alert.defaultProps.alertClose()
			expect(result).toBe(undefined)
		})

	})


})