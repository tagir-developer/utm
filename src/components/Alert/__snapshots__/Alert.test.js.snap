// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alert /> Component render. Component props testing Should Alert component render without props 1`] = `
<CSSTransition
  classNames="alert"
  in={false}
  mountOnEnter={true}
  timeout={
    Object {
      "enter": 600,
      "exit": 200,
    }
  }
  unmountOnExit={true}
>
  <div
    className="alert alert-success alert-dismissible fade show"
    role="alert"
  >
    <strong>
      Заголовок алерта
    </strong>
     
    Сообщение алерта
    <button
      className="close"
      onClick={[Function]}
      type="button"
    >
      <span
        aria-hidden="true"
      >
        ×
      </span>
    </button>
  </div>
</CSSTransition>
`;
